contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -1299,6 +1325,16 @@ exec_stmt_exec(PLtsql_execstate *estate, PLtsql_stmt_exec *stmt)
 								 GUC_ACTION_SAVE, true, 0, false);
 		SetCurrentRoleId(current_user_id, false);
 	}
+	
+	/*
+	 * Restore plan_cache_mode
+	 */
+	if ((stmt->exec_with_recompile || created_with_recompile) && !old_plan_mode)
+	{
+		(void) set_config_option("plan_cache_mode", old_plan_mode,
```

Comment:
The impact of running `is_created_with_recompile() `is 9.8% in my environment (1 million executions go from 27337 to 30027 seconds). This is the cost of a lookup for every proc call, also when RECOMPILE is not used. Is there a way to cache this attribute so as to optimize this?

The impact of setting/resetting the GUC in case of RECOMPILE is 2.3% in my environment (1 million executions go from 30273 to 30969 seconds).